apps/condo/domains/contact/components/contactRoles/BaseContactRoleForm.tsx

```diff
@@ -44,10 +45,19 @@ export const BaseContactRoleForm: React.FC<BaseTicketPropertyHintFormProps> = ({
     const ChangesSavedMessage = intl.formatMessage({ id: 'ChangesSaved' })
     const ReadyMessage = intl.formatMessage({ id: 'Ready' })
 
+    const [existingContactRoles, setExistingContactRoles] = useState<Set<string>>()
+
     const router = useRouter()
 
     const createContactRoleAction = ContactRole.useCreate({})
     const softDeleteContactRoleAction = ContactRole.useSoftDelete()
+    const { objs: contactRoles, loading } = ContactRole.useObjects({
```

Comment:
Please, create just a custom hook useExistingRolesNames and move All this logic to one place. Avoid to split the logic.

Reply:
done